mitmproxy/flowfilter.py

```diff
@@ -322,8 +322,10 @@ class FDomain(_Rex):
     flags = re.IGNORECASE
     is_binary = False
 
-    @only(http.HTTPFlow)
+    @only(http.HTTPFlow, websocket.WebSocketFlow)
     def __call__(self, f):
+        if isinstance(f, websocket.WebSocketFlow):
+            f = f.handshake_flow
```

Comment:
This is - excuse the language - a bit half-assed. Let's add a `@websocket_as_http` decorator that does this "conversion" and add it to all applicable filters.
```python
def websocket_as_http(fn):
    @functools.wraps(fn)
    def wrapped(self, flow):
        if isinstance(flow, websocket.WebSocketFlow):
            flow = flow.handshake_flow
        return fn(self, flow)
    return wrapped
```